packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -110,6 +122,26 @@ export default createRule<Options, MessageIds>({
     const checker = services.program.getTypeChecker();
     const compilerOptions = services.program.getCompilerOptions();
 
+    function getCommentDefaultCase(
+      node: TSESTree.SwitchStatement,
+    ): TSESTree.Comment | undefined {
+      const lastCase = node.cases.at(-1);
+      const commentsAfterLastCase = lastCase
+        ? context.sourceCode.getCommentsAfter(lastCase)
+        : [];
+      const defaultCaseComment = commentsAfterLastCase.at(-1);
+      const commentRegExp =
+        defaultCaseCommentPattern != null
+          ? new RegExp(defaultCaseCommentPattern, 'u')
+          : DEFAULT_COMMENT_PATTERN;
```

Comment:
[Refactor] This re-runs the `new RegExp` on each `switch` statement. Let's move it to a root statement inside the `create()`, not inside a nested function, so it only happens once.